tests/requirements/oracle.txt

```diff
@@ -1 +1 @@
-oracledb >= 1.3.2
+oracledb >= 2.3.0
```

Comment:
Sorry @felixxm I actually don't know whether there are rules around when we do these updates. 
Skimmed the [release notes](https://python-oracledb.readthedocs.io/en/latest/release_notes.html#oracledb-2-3-0-july-2024) for the supported Python versions - it seems ok. Anything that we should be wary of?

Reply:
We do them when we have to :shrug: e.g. when we want to use a new feature etc. It supports the same Python and Oracle versions as we do, so it should be painless. :+1: